### PR TITLE
Suggest parent label

### PR DESCRIPTION
**Pull request by jamesgorrie**
As a way to discover that we have related labels.
Off the back of conversations with G2 & Observer (people who use folders etc).

related-label-select 2

### CODE REVIEW

File: media-api/app/controllers/MediaApi.scala [hunk 1]
```diff
@@ -306,12 +307,29 @@ object MediaApi extends Controller with ArgoHelpers {
...
+  private val getSuggestedLabelsLink =
+    Link("suggested-labels", s"$rootUri/suggest/edits/labels")
+
   def suggestMetadataCredit(q: Option[String], size: Option[Int]) = Authenticated.async { request =>
     ElasticSearch
       .completionSuggestion("suggestMetadataCredit", q.getOrElse(""), size.getOrElse(10))
       .map(c => respondCollection(c.results))
   }
 
+  def suggestLabels() = Authenticated {
```
Comment (jamesgorrie) [inline media-api/app/controllers/MediaApi.scala]: This is where I am thinking it should be a subset of super labels. We can then autocomplete the [preset labels](https://github.com/guardian/grid/blob/master/kahuna/public/js/services/preset-label.js) too

File: media-api/app/controllers/MediaApi.scala [hunk 2]
```diff
@@ -306,12 +307,29 @@ object MediaApi extends Controller with ArgoHelpers {
...
+    val pseudoFamousLabels = List(
```
Comment (theefer) [inline media-api/app/controllers/MediaApi.scala]: Sad to hardcode in the codebase :crying_cat_face: 

Could it be an application config thing?

I'd start with it hardcoded to prove the point though, so :+1:

File: kahuna/public/js/search/results.html [hunk 3]
```diff
@@ -27,8 +27,24 @@
...
-                <div class="related-labels__title">Related to {{ctrl.parentLabel}}:</div>
+                        ng:class="{'related-labels--with-labels': ctrl.relatedLabels.length !== 0}">
+                <div class="related-labels__title">
+                    Related to:
+                    <span ng:switch="ctrl.relatedLabels.length === 0">
+                        <form class="inline-block" ng:switch-when="true" ng:submit="ctrl.setParentLabel()">
+                            <gr-datalist class="related-labels__datalist" gr:search="ctrl.labelSearch(q)">
+                                <!-- TODO: autogrow this element -->
+                                <input class="related-labels__parent" type="text"
+                                       placeholder="e.g. culture"
```
Comment (tsop14) [inline kahuna/public/js/search/results.html]: On uploads I said 'e.g. Observer'. I think we should be suggesting consistent label styles as they are case sensitive? so 'observer' and 'culture'... ?

**Comment by theefer** [comment]
The "Related to" prompt and user flow seems weird to me? Related to what? And why does selecting something in there make it search?

**Comment by theefer** [comment]
Actually looking at the screenshot the second part of the flow kind of makes sense. Might just be the "Related to __" that's odd to me. Maybe I'm wrong and @alastairjardine can pitch in.

File: kahuna/public/js/search/results.html (hunk 3 above)
Comment (jamesgorrie) [inline kahuna/public/js/search/results.html]: I think case insensitive as that's what others have been doing so far.
I'll change this to the "observer", but we could populate the example from the list later on.

File: media-api/app/controllers/MediaApi.scala (hunk 2 above)
Comment (jamesgorrie) [inline media-api/app/controllers/MediaApi.scala]: Yea, I think the point would be to have "leading" which are guardian specific, then, as they become the most popular labels, use the standard labels suggester.

Comment (theefer) [inline media-api/app/controllers/MediaApi.scala]: OK

**Comment by theefer** [comment]
:+1:

**Comment by jamesgorrie** [comment]
@theefer it is now.

**Pull request merged**